walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,106 @@
 
 package com.google.sps;
 
+import java.util.*;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    ArrayList<Event> eventsList = new ArrayList<Event>();
+    ArrayList<String> requestOptAttendeeList = new ArrayList<String>();
+    ArrayList<String> requestAttendeeList = new ArrayList<String>();
+
+    eventsList = sortList(events);
+    requestOptAttendeeList = getAttendees(request, true);
+    requestAttendeeList = getAttendees(request, false);
+
+    Collections.sort(eventsList, new eventComparator());
+
+    Collection meetingTimes = new ArrayList<TimeRange>();
+    Collection meetingOptTimes = new ArrayList<TimeRange>();
+
+    /* startTime(2) and endTime(2) are used to adjust the time range for each available meeting slot
+    with duration being the length of the meeting */
+    int startTime = 0;
+    int startTime2 = 0;
+    int endTime = 0;
+    int endTime2 = 0;
+    int endOfDayTime = 24*60 - 1;
+    long duration = request.getDuration();
+
+    for (int i = 0; i < eventsList.size(); i++){
+      if (listSimilarity(requestAttendeeList, eventsList.get(i).getAttendees()) == true) {
+        int meetingStartTime = eventsList.get(i).getWhen().start();
+        int meetingDuration = eventsList.get(i).getWhen().duration();
+        
+        // Checks to see if there's any time left for a meeting
+        if (meetingStartTime - startTime >= duration){
+          endTime = meetingStartTime - 1;
+          meetingTimes.add(TimeRange.fromStartEnd(startTime, endTime, true));
+        }
+        if(startTime < meetingStartTime + meetingDuration){ startTime = meetingStartTime + meetingDuration;}
+        if (startTime == 1440){startTime--;}
```

Comment:
Shouldn't this be instead:

if (startTime > endOfDayTime) { startTime = endOfDayTime; }